src/index.d.ts

```diff
@@ -60,34 +60,33 @@ export interface ContractArtifact {
 export declare const BatchExchangeArtifact: ContractArtifact;
 export declare const BatchExchangeViewerArtifact: ContractArtifact;
 
-export interface Order {
+export interface Order<T = string> {
```

Comment:
Why not explicitly use string in the places where string is used? Isn't this just unnecessary obfuscation?

Reply:
Its because we need `Order<string>` for `decodeOrders` and `Order<BN>` for `decodeOrdersBN` later in the file. Also, the streamed orderbook uses `Order<bigint>`.

Reply:
Generics 😱 

Hopefully we can get rid of the duplicate functions then when we move everything to ts.